src/RestApi/StoreApi/Controllers/Customer.php

```diff
@@ -0,0 +1,161 @@
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\RestApi\StoreApi\Controllers;
+
+defined( 'ABSPATH' ) || exit;
+
+use \WP_Error as RestError;
+use \WP_REST_Server as RestServer;
+use \WP_REST_Controller as RestController;
+use \WC_Customer as CustomerObject;
+use Automattic\WooCommerce\Blocks\RestApi\StoreApi\Schemas\CustomerSchema;
+
+/**
+ * Customer API.
+ */
+class Customer extends RestController {
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/store';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'customer';
+
+	/**
+	 * Schema class instance.
+	 *
+	 * @var object
+	 */
+	protected $schema;
+
+	/**
+	 * Setup API class.
+	 */
+	public function __construct() {
+		$this->schema = new CustomerSchema();
+	}
+
+	/**
+	 * Register routes.
+	 *
+	 * @todo /session
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			[
+				[
+					'methods'  => RestServer::READABLE,
+					'callback' => [ $this, 'get_item' ],
+					'args'     => [
+						'context' => $this->get_context_param( [ 'default' => 'view' ] ),
+					],
+				],
+				[
+					'methods'  => RestServer::EDITABLE,
+					'callback' => array( $this, 'update_item' ),
+					'args'     => $this->get_endpoint_args_for_item_schema( RestServer::EDITABLE ),
+				],
+				'schema' => [ $this, 'get_public_item_schema' ],
+			]
+		);
+	}
+
+	/**
+	 * Get the current customer.
+	 *
+	 * @param \WP_REST_Request $request Full details about the request.
+	 * @return \WP_Error|\WP_REST_Response
+	 */
+	public function get_item( $request ) {
+		$customer = wc()->cart->get_customer();
+
+		if ( ! $customer || ! $customer instanceof CustomerObject ) {
+			return new RestError(
+				'woocommerce_rest_customer_error',
+				__( 'Unable to retrieve customer.', 'woo-gutenberg-products-block' ),
+				[ 'status' => 500 ]
+			);
+		}
+
+		return $this->prepare_item_for_response( $customer, $request );
+	}
+
+	/**
+	 * Update the current customer.
+	 *
+	 * @param \WP_Rest_Request $request Full data about the request.
+	 * @return \WP_Error|\WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$customer = wc()->cart->get_customer();
+
+		if ( ! $customer || ! $customer instanceof CustomerObject ) {
+			return new RestError(
+				'woocommerce_rest_customer_error',
+				__( 'Unable to retrieve customer.', 'woo-gutenberg-products-block' ),
+				[ 'status' => 500 ]
+			);
+		}
+
+		try {
+			if ( isset( $request['billing'] ) ) {
+				$allowed_billing_values = array_intersect_key( $request['billing'], ( $this->get_item_schema() )['properties']['billing']['properties'] );
+				foreach ( $allowed_billing_values as $key => $value ) {
+					$customer->{"set_billing_$key"}( $value );
+				}
+			}
```

Comment:
Are `$request['billing']` fields typically controlled via payment methods or are those standard fields Woo Core displays during checkout if required by a payment method (via indicating it needs billing info)?  The question isn't regarding whether this code should be here or not, but to gain understanding for how this impacts how payment methods will integrate with the new checkout.

I'm guessing this is something core displays in the checkout if billing info is required.  So for payment methods in the new checkout, we probably just need them to flag they need billing info?

Reply:
They are a 'standard' set in core, but extensible. Not sure how widely used the extensibility is.

It would probably be boolean off/on for payment methods, but we'd need to be careful that this data wasn't needed for other things, for example, tax liability, additional verification/trust scores..more?

Reply:
Okay, so then for the new checkout, I think we'll want the billing field step to be controlled via checkout.  Payment Methods can force that step to be on (if needed) but it can also be controlled via checkout settings.